core/store/src/flat/store_helper.rs

```diff
@@ -174,3 +174,73 @@ pub fn key_belongs_to_shard(
     let (key_shard_uid, _) = decode_flat_state_db_key(key)?;
     Ok(key_shard_uid.version == shard_layout.version() && key_shard_uid.shard_id as u64 == shard_id)
 }
+
+#[cfg(test)]
+mod tests {
+    use super::iter_flat_state_entries;
+    use crate::flat::store_helper::{set_delta, set_flat_state_value};
+    use crate::flat::{
+        BlockInfo, FlatStateChanges, FlatStateDelta, FlatStateDeltaMetadata, FlatStateValue,
+        FlatStorageManager,
+    };
+    use crate::test_utils::create_test_store;
+    use crate::CryptoHash;
+    use near_primitives::shard_layout::ShardUId;
+
+    #[test]
+    fn test_iter_flat_state_entries() {
+        // Setup shards and store
+        let store = create_test_store();
+        let shard_uids = Vec::from([
+            ShardUId { shard_id: 0, version: 0 },
+            ShardUId { shard_id: 1, version: 0 },
+            ShardUId { shard_id: 2, version: 0 },
+        ]);
+        let root = CryptoHash::default();
+        let flat_storage_manager = FlatStorageManager::test(store.clone(), &shard_uids, root);
+
+        for (i, shard_uid) in shard_uids.iter().enumerate() {
+            let mut store_update = store.store_update();
+            let key: Vec<u8> = vec![0, 1, i as u8];
```

Comment:
Is it fine that `key` has odd-length?
Does `key` need to have `i`? Can different shards be tested with the same `key` and `val`?

Reply:
Not sure about the `key` from a general point of view, but I think from this context yes since we are avoiding the semantics of `key` since we are not dealing with trie, this is just a test for `iter_flat_state_entries` to see if after assigning values that we can actually iterate.
No `key` does not have to have `i` it can be the same for both, but with this, we also verify that the correct shard has a correct key-value pair.